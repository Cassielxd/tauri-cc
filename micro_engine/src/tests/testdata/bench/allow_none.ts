import { unreachable } from "../../../../test_util/std/testing/asserts.ts";

const permissions: Deno.PermissionName[] = [
  "read",
  "write",
  "net",
  "env",
  "run",
  "ffi",
  "hrtime",
];

for (const name of permissions) {
  Deno.bench({
    name,
    permissions: {
      [name]: true,
    },
    fn() {
      unreachable();
    },
  });
}
