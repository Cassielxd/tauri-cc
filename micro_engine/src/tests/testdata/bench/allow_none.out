Check [WILDCARD]/bench/allow_none.ts
cpu: [WILDCARD]
runtime: deno [WILDCARD] ([WILDCARD])

[WILDCARD]/bench/allow_none.ts
benchmark      time (avg)        iter/s             (min â€¦ max)       p75       p99      p995
--------------------------------------------------------------- -----------------------------
read       error: PermissionDenied: Can't escalate parent thread permissions
[WILDCARD]
write      error: PermissionDenied: Can't escalate parent thread permissions
[WILDCARD]
net        error: PermissionDenied: Can't escalate parent thread permissions
[WILDCARD]
env        error: PermissionDenied: Can't escalate parent thread permissions
[WILDCARD]
run        error: PermissionDenied: Can't escalate parent thread permissions
[WILDCARD]
ffi        error: PermissionDenied: Can't escalate parent thread permissions
[WILDCARD]
hrtime     error: PermissionDenied: Can't escalate parent thread permissions
[WILDCARD]
error: Bench failed
